src/index.ts

```diff
@@ -1,9 +1,35 @@
-import * as lib from "./worker";
+import { hasProperties } from "./util/object";
+import loadWhenDocumentReady from "./util/loadWhenDocumentReady";
+import Worker from "workerize-loader!./worker";
 
-if ("Worker" in window && "Promise" in window) {
-  const worker = (lib as any)() as typeof lib;
+// List of required features browser features
+const requiredFeatures = ["Worker", "Promise", "fetch"];
+// Test whether browser has required feature support
+const hasFeatureSupport = hasProperties(window, requiredFeatures);
```

Comment:
Do you expect to use `hasProperties` for other things in the future? Writing a util seems a little overkill if something like this works:

```
let has = false;
try {
has = window.Worker && window.Promise && window.fetch
}
```

Reply:
Good question. Yes, we will use the `hasProperties` util within the main library as we do some feature detection for the various `window.perfromance` sub APIs. 

We no longer use it here, but the other part of this util is that it accepts a object path as a string` i.e. `my.nested.object`, which is useful when testing the deeper performance apis mentioned above.